### PR TITLE
Chore/remove watt form field

### PR DESCRIPTION
**Pull request by mimse**
Clean up in our legacy components.

Replaced all our old watt-form-field with watt-field

Also sorry for the major PR, but is was cascading changes :-(

## References

- https://github.com/Energinet-DataHub/greenforce-frontend/issues/1930

### CODE REVIEW

File: libs/dh/charges/feature-create-prices/src/lib/dh-charges-create-prices.component.html [hunk 1]
```diff
@@ -23,108 +23,117 @@ <h3>{{ t("createPrices.title") }}</h3>
...
-              *ngIf="charge.controls['effectiveDate'].hasError('maxDate') && hasSubmitted"
-            >
-              {{ t("createPrices.maxDateValidationErrorText") }}
-            </watt-error>
+              <watt-field-error
+                *ngIf="charge.controls['effectiveDate'].hasError('maxDate') && hasSubmitted"
+              >
+                {{ t("createPrices.maxDateValidationErrorText") }}
+              </watt-field-error>
+            </watt-datepicker>
```
Comment (dzhavat) [inline libs/dh/charges/feature-create-prices/src/lib/dh-charges-create-prices.component.html:136]: issue: validation errors are shown immediately after opening the page. Should be hidden until the user has interacted with the form, or submitted the form.

image

File: libs/watt/src/lib/components/stepper/+storybook/stepper.example.component.html [hunk 2]
```diff
@@ -17,16 +17,27 @@
...
+      <watt-text-field
+        class="firstname"
+        [formControl]="user.controls.firstname"
+        type="text"
+        label="firstname"
+      >
+        <watt-field-error *ngIf="user.controls.firstname.hasError('required')"
+          >Error</watt-field-error
+        >
+      </watt-text-field>
```
[the remark this answers is not in the document]
Comment (mimse) [reply libs/watt/src/lib/components/stepper/+storybook/stepper.example.component.html:29]: Going to make a bug on this one

https://github.com/Energinet-DataHub/team-titans/issues/286

File: libs/dh/wholesale/feature-calculations/src/lib/create/create.component.html [hunk 3]
```diff
@@ -106,44 +105,42 @@
...
-            <watt-error *ngIf="formGroup.controls.dateRange.errors?.rangeRequired">
+          <watt-datepicker
+            [label]="t('create.period.label')"
+            formControlName="dateRange"
+            [rangeMonthOnlyMode]="monthOnly.includes(formGroup.controls.processType.value)"
+            [range]="true"
+            [min]="minDate"
+            [max]="maxDate"
+          >
+            <watt-field-error *ngIf="formGroup.controls.dateRange.errors?.rangeRequired">
```
[the remark this answers is not in the document]
Comment (mimse) [reply libs/dh/wholesale/feature-calculations/src/lib/create/create.component.html:116]: Should be fixed

File: libs/dh/metering-point/feature-search/src/lib/form/dh-metering-point-search-form.component.html [hunk 4]
```diff
@@ -19,29 +19,26 @@
...
     <watt-button
       icon="close"
       type="reset"
       variant="icon"
-      wattSuffix
       data-testid="buttonClear"
       (click)="onSearchInputClear()"
     ></watt-button>
-  </watt-form-field>
+  </watt-text-field>
```
[the remark this answers is not in the document]
Comment (mimse) [reply libs/dh/metering-point/feature-search/src/lib/form/dh-metering-point-search-form.component.html:40]: Are going to be removes soon

File: libs/dh/charges/feature-create-prices/src/lib/dh-charges-create-prices.component.html (hunk 1 above)
Comment (mimse) [reply libs/dh/charges/feature-create-prices/src/lib/dh-charges-create-prices.component.html:136]: Should be fixed

**Review by frontend-specialisten — APPROVED** [review]

**Pull request merged**